editor/components/post-author/index.js

```diff
@@ -5,45 +5,107 @@ import { __ } from '@wordpress/i18n';
 import { withInstanceId } from '@wordpress/components';
 import { Component, compose } from '@wordpress/element';
 import { withSelect, withDispatch } from '@wordpress/data';
+import apiRequest from '@wordpress/api-request';
 
 /**
  * Internal dependencies
  */
 import PostAuthorCheck from './check';
 
+/**
+ * External dependencies
+ */
+import Downshift from 'downshift';
+import { debounce } from 'underscore';
+
 export class PostAuthor extends Component {
 	constructor() {
 		super( ...arguments );
 
 		this.setAuthorId = this.setAuthorId.bind( this );
+		this.suggestAuthors = this.suggestAuthors.bind( this );
 	}
 
-	setAuthorId( event ) {
+	setAuthorId( selected ) {
 		const { onUpdateAuthor } = this.props;
-		const { value } = event.target;
-		onUpdateAuthor( Number( value ) );
+		const { id } = selected;
+		onUpdateAuthor( Number( id ) );
+	}
+
+	suggestAuthors( query, args ) {
+		if ( ! args.isOpen ) {
+			return;
+		}
+		const payload = '?search=' + encodeURIComponent( query );
+		this.setState( { searching: true } );
+		apiRequest( { path: '/wp/v2/users' + payload } ).done( ( results ) => {
```

Comment:
Should we leverage the data module instead? We already have a `users` reducer their which I believe should be refactored to be an entity and build support to searching/pagination into the entities. related #6395